File: homework/theBoldGuy/insertion_sort.py
def insertion_sort(arr, low, high):
    sorted_arr=arr[:]
    for i in range(low+1, high+1):
        key = sorted_arr[i]
        j = i - 1
        while j >= low and sorted_arr[j] > key:
            sorted_arr[j+1] = sorted_arr[j]
            j -= 1
        sorted_arr[j+1] = key
    return sorted_arr

arr = [45, 15, 0, -4, 76, 888, 345]
sorted_arr = insertion_sort(arr, 0, len(arr) - 1)
print(f'Base array: {arr}')
print(f'Sorted array: {sorted_arr}')

File: homework/theBoldGuy/insertion_sort.dart
List<String> insertionSort(List<String> arr, low, high) {
  List<String> sortedArr =  List.from(arr);

  for (int i = low +1; i <= high; i++) {
    String key = sortedArr[i];
    int j = i - 1;

    while (j >= low && sortedArr[j].compareTo(key) > 0) {
      sortedArr[j+1] = sortedArr[j];
      j--;
    }
    sortedArr[j+1] = key;
  }
  return sortedArr;
}

void main() {
  List<String> arrMain = ['c', 'k', 'books', '-4', 'fire', 'cat', 'align'];
  List<String> sortedArrMain = insertionSort(arrMain, 0, arrMain.length - 1);

  print('Base array: $arrMain');
  print('Sorted array: $sortedArrMain');
}
